### ISSUE TITLE
Completions after isolated keyword are not intelligent enough

### ISSUE THREAD

**Issue by nadeeshaan** [ballerina-platform/ballerina-lang#30685]
**Description:**
Consider the following source snippets

`public type  Hello isolated <cursor>`

```
public type  Hello isolated <cursor>

function hello() {}
```

In both of the cases, the particular context identified, is not consistent

**Comment by lochana-chathura** [comment]
This is fixed in master. They both recover the same way.

**Comment by lochana-chathura** [comment]
Need to add a parser test case and then close this issue.

**Comment by lochana-chathura** [comment]
@nadeeshaan in both cases, context needs to be TypeDefinitionNodeContext right. 
So [2] is okay. Need to improve [1] right?
[1]
```bal
public type  Hello isolated <cursor>
```

[2] 
```bal
public type  Hello isolated <cursor>

function hello() {}
```

**Comment by lochana-chathura** [comment]
@nadeeshaan with the linked PR, parsing is fixed. But need some LS changes on top of this. 
i.e. For the above sample, only `function` keyword is suggested now. 

So are we going to keep this issue for the LS changes or I can close this with the parser changes?

**Comment by nadeeshaan** [comment]
> @nadeeshaan with the linked PR, parsing is fixed. But need some LS changes on top of this.
> i.e. For the above sample, only `function` keyword is suggested now.
> 
> So are we going to keep this issue for the LS changes or I can close this with the parser changes?

With the attached PR, is it possible to access the associated type qualifiers without traversing through the minutiae list and yes we have to keep this issue open until the LS changes are done.

**Comment by lochana-chathura** [comment]
> With the attached PR, is it possible to access the associated type qualifiers without traversing through the minutiae list

yes. It is now recovering as `public type Hello isolated MISSING[function] MiSSING[;]` so the qualifier is not in the minutiea.

> we have to keep this issue open until the LS changes are done.

Ack. will unlink the pr from this, issue and create a separate issue for parser changes.

**Comment by lochana-chathura** [comment]
> Ack. will unlink the pr from this issue and create a separate issue for parser changes.

Created #31406

**Comment by mohanvive** [comment]
Related issues: https://github.com/ballerina-platform/ballerina-lang/issues/26017

**Comment by malinthar** [comment]
Blocked by https://github.com/ballerina-platform/ballerina-lang/issues/32445
